pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -522,7 +523,7 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPRouteConfig(env mo
 		}
 
 		for _, svc := range guardedHost.Services {
-			domains := generateAltVirtualHosts(svc.Hostname, guardedHost.Port)
+			domains := generateAltVirtualHosts(svc.Hostname, guardedHost.Port, svc.MeshExternal)
```

Comment:
Don’t call this function if mesh external is set.

Reply:
the function returns hostname:port that is needed in domains.

Reply:
YEs, but you don't need such a big function for that small thing. if mesh External, just add hostname, and hostname+port, else call generateAltVirtualHosts(), which keeps its logic generic to a given name+port. We could then reuse the same function elsewhere or move it to util

Reply:
Was thinking about doing that. Would it be better to move the first line of generateAltVirtualHosts() out, and put it in here? And don't call the function if it's mesh external?